xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -399,13 +399,32 @@ actor UserProfileService {
         let following = try await self.getFollowingList(
             address: address
         )
-        let notes = try await sphere.list()
         let followingStatus = await self.addressBook.followingStatus(did: did)
         
-        let entries = try await self.loadEntries(
-            address: address,
-            slugs: notes
-        )
+        let entries = try await Func.run {
+            switch (followingStatus) {
+            // Read from DB if we follow this user
+            case .following(let name):
+                return try
+                    self.database.listRecentMemos(owner: did)
```

Comment:
Noting that it is technically possible to JOIN tables and map the DID to a petname, returning the petname with the data.

However, I do not think we should worry about this rn.